src/foundation.cs

```diff
@@ -8752,7 +8753,7 @@ partial interface NSItemProvider : NSCopying {
 
 		[Mac (10,12)][Async]
 		[Export ("registerCloudKitShareWithPreparationHandler:")]
-		void RegisterCloudKitShare ([BlockCallback] Action<CloudKitRegistrationPreparationHandler> preparationHandler);
+		void RegisterCloudKitShare (CloudKitRegistrationPreparationAction preparationHandler);
```

Comment:
@chamons I think your fix is right, to solve the breaking change you'll just need to add a method (under `!XAMCORE_4_0`) with the old signature (that calls the new one).